### PR TITLE
dotCMS/core#25693 Define minimum data to show bayesian chart

### LINKED ISSUE #25693: [UI] Define the the minimum of data required to show Bayesian Chart
### Parent Issue

#24343

### User Story

As a user, I want to see the chart with enough reliable data gathered to show it.

### Acceptance Criteria
- Define the minimum data required to show a reliable chart.

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Define minimum data to show bayesian chart
* change 2

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### <samp>🤖 Generated by Copilot at e8cc43f</samp>

This pull request improves the experiments reports feature by adding a bayesian chart, fixing some bugs, and cleaning up the code. It also changes the proxy configuration file for the development server from JSON to MJS to enable dynamic imports. The main files affected are `dot-experiments-reports.component.html`, `dot-experiments-reports.component.ts`, `dot-experiments-reports-store.ts`, `dot-experiment.utils.ts`, `project.json`, and `proxy-dev.conf.mjs`.

## Related Issue
Fixes #25693 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at e8cc43f</samp>

*  Changed the proxy configuration file for the development server from a JSON file to an MJS module file to allow ES modules syntax and dynamic imports ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-a5a7709ae811f21c30177ca89ea61196bd970f2695ce9a24aee461b0e792dfa8L101-R101), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-f162544d3441baeebb73cbea3b36673232b11068a7307feb36347b6f2c7ad842R1-R19))
* Added a constant value for the minimum number of sessions required to show the experiments chart and used it to determine if the chart data is empty or not ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-f1a1a6a6f21372c106dd50983b66d511523dd176656f6d368fe830a87a26b09cR18-R19), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL103-R141), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL141-R156))
* Defined an interface for the chart data and the flag to indicate if the chart has enough data to be displayed and used it for the daily and bayesian charts ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL59-R73), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL252-R283))
* Added conditions to check the suggested winner status and the failure count of the bayesian result and return empty data arrays if there is no clear winner or no data to generate the probability density function ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-da9b3c84b5b9a7d65edb918c808e66eb2594cd7ef814e2e65bb9a641a166fc44L165-R171), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-da9b3c84b5b9a7d65edb918c808e66eb2594cd7ef814e2e65bb9a641a166fc44L174-R181))
* Added a step parameter to the generateProbabilityDensityData function to control the increment of the x values in the data array and replaced the constant value with the parameter ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-da9b3c84b5b9a7d65edb918c808e66eb2594cd7ef814e2e65bb9a641a166fc44L191-R204), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-da9b3c84b5b9a7d65edb918c808e66eb2594cd7ef814e2e65bb9a641a166fc44L203-R219))
* Removed the condition to show the experiments summary from the template and always showed the summary regardless of the experiment status ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-3aa1915a72f87db98f07027a724d6943479b6ac919e63e429b70cab1c7b7fee8L12))
* Changed the data input for the daily and bayesian chart components from the chart data properties to the chart objects of the view model ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-3aa1915a72f87db98f07027a724d6943479b6ac919e63e429b70cab1c7b7fee8L25-R25), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-3aa1915a72f87db98f07027a724d6943479b6ac919e63e429b70cab1c7b7fee8L37-R37))
* Added two preformatted elements to the template for debugging purposes to display the JSON representation of the bayesianChart property and the hasEnoughSessions property of the view model ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-3aa1915a72f87db98f07027a724d6943479b6ac919e63e429b70cab1c7b7fee8R52-R59))
* Imported and provided the JsonPipe in the component file and used it to transform an object into a JSON string in the template ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-b78882e82a52f85a9be72e848d99f3389077b9843e1ea1f9f8e62b89222741baR5), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-b78882e82a52f85a9be72e848d99f3389077b9843e1ea1f9f8e62b89222741baL60-R62))
* Added the dotMessageService property to the constructor parameters of the component class and used it to get the localized messages for the chart labels ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-b78882e82a52f85a9be72e848d99f3389077b9843e1ea1f9f8e62b89222741baL86-R85))
* Added the setComponentStatus and setExperiment updaters to the store class and used them to update the status and experiment properties of the state ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeR176-R192))
* Imported the BayesianStatusResponse enum from the constants file and used it to check the suggested winner status of the bayesian result ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL16-R25))
* Removed the inject function, the dotMessageService property and the showExperimentSummary$ observable from the component and store files as they were not used and probably leftovers from a previous refactor ([link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-b78882e82a52f85a9be72e848d99f3389077b9843e1ea1f9f8e62b89222741baL11-R12), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-b78882e82a52f85a9be72e848d99f3389077b9843e1ea1f9f8e62b89222741baL69-R74), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL241-R273), [link](https://github.com/dotCMS/core/pull/25799/files?diff=unified&w=0#diff-84492bfda255aebc3ac5eb5bfb8820dda17da27f44299146f30570f5bed75cfeL265-R295))

### CODE REVIEW

File: core-web/apps/dotcms-ui/project.json [hunk 1]
```diff
@@ -98,7 +98,7 @@
             "executor": "@angular-devkit/build-angular:dev-server",
             "options": {
                 "browserTarget": "dotcms-ui:build:development",
-                "proxyConfig": "apps/dotcms-ui/proxy-dev.conf.json"
+                "proxyConfig": "apps/dotcms-ui/proxy-dev.conf.mjs"
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/project.json:101]: Remove the file `proxy-dev.conf.json`

File: core-web/apps/dotcms-ui/proxy-dev.conf.mjs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export default [
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/proxy-dev.conf.mjs:1]: Nice!

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts [hunk 3]
```diff
@@ -159,6 +173,23 @@ export class DotExperimentsReportsStore extends ComponentStore<DotExperimentsRep
...
     );
 
+    // Updaters
+    readonly setComponentStatus = this.updater(
+        (state: DotExperimentsReportsState, status: ComponentStatus) => ({
+            ...state,
+            status
+        })
+    );
+    readonly setExperiment = this.updater(
```
Comment (fmontes) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts:183]:
```suggestion

    readonly setExperiment = this.updater(
```

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts [hunk 4]
```diff
@@ -238,9 +269,8 @@ export class DotExperimentsReportsStore extends ComponentStore<DotExperimentsRep
         this.state$,
         this.isLoading$,
         this.hasEnoughSessions$,
-        this.showExperimentSummary$,
-        this.getDailyChartData$,
-        this.getBayesianChartData$,
+        this.bayesianChart$,
+        this.dailyChart$,
```
Comment (fmontes) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts:273]: This keeps getting bigger and bigger... no way to split?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/dot-experiment.utils.ts [hunk 5]
```diff
@@ -188,27 +194,59 @@ export const getBayesianDatasets = (
...
+    const referenceSlope = calculateSlope(points[0], points[1]);
+
+    for (let i = 1; i < points.length - 1; i++) {
+        const slope = calculateSlope(points[i], points[i + 1]);
+        if (Math.abs(slope - referenceSlope) > 1e-6) {
+            return false; // Points do not form a line
+        }
+    }
+
+    return true;
```
Comment (fmontes) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/dot-experiment.utils.ts:251]:
```suggestion
const arePointsALine = (points: { x: number; y: number }[]): boolean => {
    if (points.length < 3) {
        return true;
    }

    const referenceSlope = (points[1].y - points[0].y) / (points[1].x - points[0].x);

    return points.slice(1).every((point, i) => {
        const slope = (point.y - points[i].y) / (point.x - points[i].x);
        return Math.abs(slope - referenceSlope) < 1e-6;
    });
};
```

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.spec.ts [hunk 6]
```diff
@@ -548,4 +523,188 @@ describe('DotExperimentsReportsStore', () => {
...
+                                }
+                            }
+                        }
+                    },
+                    sessions: { total: 20, variants: { DEFAULT: 10, '111': 10 } },
+                    bayesianResult: {
+                        ...EXPERIMENT_MOCK_RESULTS.bayesianResult,
+                        suggestedWinner: BayesianStatusResponse.NONE
+                    }
+                })
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.spec.ts:566]: I see that you are repeating this object through all this file and just changing one section. Should we do a builder function for this object?

```
getMockedResult({
                   count: 8,
                   totalPercentage: 100.0,
                   variantPercentage: 100.0
                });
```

Something like that.

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts [hunk 7]
```diff
@@ -100,33 +103,39 @@ export class DotExperimentsReportsStore extends ComponentStore<DotExperimentsRep
...
 
-    readonly showExperimentSummary$: Observable<boolean> = this.select(({ experiment }) =>
-        Object.values([
-            DotExperimentStatus.ENDED,
-            DotExperimentStatus.RUNNING,
-            DotExperimentStatus.ARCHIVED
-        ]).includes(experiment?.status)
+    // Daily Chart
+    readonly hasEnoughSessions$: Observable<boolean> = this.select(
+        ({ results }) => results != null && results.sessions.total >= MINIMUM_SESSIONS_TO_SHOW_CHART
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts:138]: The assertion to null is not needed `result && ...`

**Review by zJaaal — COMMENTED** [review]
See my suggestions

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts (hunk 7 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts:138]: done

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**